packages/editor/src/browser/editor.ts

```diff
@@ -33,6 +33,14 @@ export interface TextEditorDocument extends lsp.TextDocument, Saveable, Disposab
      * @since 1.8.0
      */
     findMatches?(options: FindMatchesOptions): FindMatch[];
+    /**
+     * Create a valid position.
+     */
+    validatePosition(position: Position): Position;
```

Comment:
The name sounds like it would throw an exception for non-valid position. I would prefer `toValidPosition`.  Also, documentation of how it will or should adjust the position would be nice.

Reply:
https://github.com/eclipse-theia/theia/pull/11984/commits/96954d28fc72319d5e6b179b80b3789adc311899